Crumbling floor

ColeKincaid3 opened the pull request:
#28 
I created a crumbling floor script that uses the onTriggerEnter2d,onTriggerStay2d, and onTriggerExit2d to trigger audio and destruction of the platform, simulating the floor has crumbled. The warning audio will play each time you step on the platform and 1 second after the player has collided with the platform then the crumble audio begins and when the audio ends the platform is destroyed. 

There is a scene called CrumblingFloorTest where there is a simple easy platforming thing to try it out on. 

I also created a prefab of the platform.

ColeKincaid3 committed:
added crumbling floor prefab

CrumblingFloorEvent script is attached to child gameobject of the crumbling floor, when collision is triggered by a player tagged object there is a delay, which can be changed in the inspector, then the parent crumblingfloor gameobject is destroyed.

ColeKincaid3 committed:
Added Sound Effect upon collision with platform

-setting the delay of the crumble to 1.35 syncs well with the audio clip provided

ColeKincaid3 committed:
Merge from main into crumbling floor

ColeKincaid3 committed:
Updated Crumbling Floor Script

Now the platform triggers the warning sound when jumped on then begins the crumbling (destruction of the platform) when the player has been on the platform for 1 second.

Currently trying to fix the audio for when the floor begins to fall apart because at the moment it spamming the crumblingfloor audio clip while touching the platform during the crumbling phase.

ColeKincaid3 committed:
Update CrumblingFloorEvent.cs

Fixed the audio issue by creating a boolean that is false if the crumbling audio has not been played. if it is true when you stay on the platform and if the boolean is true then the audio will not play again.

ColeKincaid3 committed:
Merge branch 'main' into CrumblingFloor

BUNN1E5 reviewed (commented):
It looks good, I don't think the rigid bodies are required though.

ColeKincaid3 committed:
Merge from main

Kainkun commented on Assets/Scripts/CrumblingFloorEvent.cs:
```diff
@@ -0,0 +1,67 @@
+        }
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
```
use [OnCollision](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnCollisionEnter2D.html) stuff instead of ontrigger. then you can get rid of the unnecessary child with the trigger.

Kainkun commented on Assets/Scripts/CrumblingFloorEvent.cs:
```diff
@@ -0,0 +1,67 @@
+    {
+        if (collision.gameObject.tag == "Player")
+        {
+            Debug.Log("player has come in contact with the platform");
```
id appreciate it if you comment out the print statements before you do a pull request so it doesn't clutter up the debug log for others.

Kainkun commented on Assets/Scripts/CrumblingFloorEvent.cs:
```diff
@@ -0,0 +1,67 @@
+
+    private void Update()
+    {
+        if (isColliding == true)
```
Sorry that I wasn't more specific but the rocks should crumble if collided even if you're no longer on it anymore. Right now you can jump over and over and have the platform not disappear.
For example, if I just on the platform and then jump off of it, it should still disappear after the given time

Kainkun reviewed (changes requested):
Hey, good work! There are some changes I'd like for you to make. You can check out the comments in the code

ColeKincaid3 committed:
Merge branch 'main' into CrumblingFloor

ColeKincaid3 committed:
updated CrumblingFloor Structure and script

-delete old prefab and made a new simpler prefab
   *no longer uses OnTriggerEnter2d so prefab is simplified to only one gameobject
-deleted Crumbling FloorEvent script
   *changed from using OnTrigger functions to using OnCollision functions
-platform crumbles one second after player collides with platform rather than if the player has been in contact with the platform for one second
-Removed old test scene and made scene "crumblingfloor2.0" as a scene to test the prefab

ColeKincaid3 committed:
update from main

ColeKincaid3 committed:
Commented out debug.log

-making sure the debug log is not being filled by script

ColeKincaid3 commented:
I created new versions of the crumbling floor script, prefab and a new test scene called "crumblingfloor2.0". The script now uses OnCollision functions and no longer clutters the debug window with debug.log comments.

Kainkun committed:
Merge branch 'main' into CrumblingFloor

Kainkun committed:
fixed prefab

Kainkun reviewed (approved):
Work great, Thanks!

Kainkun committed:
meta
